File: Unit01/Fruit.java
public class Fruit {
    private String name;
    private double price;
    private String color;

    Fruit (String name, double price, String color) {
        this.name = name;
        this.price = price;
        this.color = color;
    }

    void discount (int percent) {
        this.price = this.price * (percent / 100.0);
    }

    public String getName () {
        return this.name;
    }

    public double getPrice () {
        return price;
    }

    public void setPrice (double price) {
        this.price = price;
    }

    public String getColor () {
        return color;
    }

    @Override
    public String toString () {
        return "A " + this.color + " " + this.name + " for $" + this.price;
    }

    @Override
    public boolean equals (Object other) {
        if (other instanceof Fruit) {
            Fruit f = (Fruit)other;
            return this.name.equals (f.name) && this.color.equals(f.color);
        } else {
            return false;
        }
   }
    
public static void main (String[] args) {
    Fruit apple1 = new Fruit ("Apple", 2.0, "Red");

    apple1.discount (50);

    System.out.println(apple1);

    // Fruit orange = new Fruit ("Orange", 1.25, "Orange");
    // System.out.println(orange);

    Fruit apple2 = new Fruit ("Apple", 2.0, "Red");

    apple2.discount (50);

    System.out.println(apple2);

    System.out.println(apple1 == apple2);
    System.out.println (apple1.equals(apple2));
    System.out.println (apple1.equals(apple1));
    
    }

}

File: Unit01/fruit.py
# Alex Tedesco

class Fruit:
    '''A class named fruit'''
    __slots__ = ["__name", "__price", "__color"]
    def __init__(self, name, price, color):
        self.__name = name
        self.__price = price
        self.__color = color

    def discount (self, percent):
        self.__price = self.__price * (percent / 100)

    def get_name (self):
        return self.__name

    def get_price (self):
        return self.__price
    
    def get_color (self):
        return self.__color

    def __str__ (self):
        # A Red Apple for $1.25
        return "A " + self.__color + " " + self.__name + " $" + str (self.__price)


    def __eq__(self, other):
        if type (self) == type (other):
            return self.__name == other.__name and self.__color == other.__color
        return False

def main ():
    apple = Fruit ("Apple", 1.25, "Red")
    # apple.name = "Apple"
    # apple.price = 1.25
    # apple.color = "Red"

    orange = Fruit ("Orange", "2.00", "Orange")
    # orange.name = "Orange"
    # orange.price = 2.00
    # orange.color = "Orange"

    # print(apple.get_name (), apple.get_price (), apple.get_color ())
    # print(orange.get_name (), orange.get_price (), orange.get_color ())
    print (apple)
    print (orange)

    apple.discount (25)
    # print (apple.get_price ())
    # print (orange.get_price ())

    print (apple == orange)
    print (apple == "Hello!")

if __name__ == "__main__":
    main ()